src/sandbox/http/invoke-lambda.js

```diff
@@ -1,93 +1,140 @@
-  let runtime = path.join(__dirname, '/runtimes/node.js')
-  let command = 'node'
-  let script = fs.readFileSync(runtime).toString()
-  let options = ['-e', minify(script)]
-  let args = {shell:true, cwd, env}
-
-  // select the command based on the current function runtime (look at .arc then at .arc-config for override)
-  let config = exists(path.join(cwd, '.arc-config'))
-  if (config) {
-    let raw = fs.readFileSync(path.join(cwd, '.arc-config')).toString()
-    let arc = parse(raw)
-    let runtime = arc.aws.find(tuple=> tuple.includes('runtime'))
-    if (runtime && runtime[1] === 'python3.6') {
-      runtime = path.join(__dirname, '/runtimes/python.py')
-      command = 'python3'
-      script = `"${fs.readFileSync(runtime).toString()}"`
-      options = ['-c', script]
+    switch (runtime) {
+      case 'python3.6':
+        runInPython(options, timeout, callback)
+        break
+      default:
+        runInNode(options, timeout, callback)
     }
+  }
+}
 
-    let ttl = arc.aws.find(tuple=> tuple.includes('timeout'))
-    // valid ttl is between 3 and 900 seconds (15 minutes)
-    if (ttl && ttl[1] > 3 && ttl[1] < 900) {
-      timeout = ttl[1]*1000 // ttl seconds in milliseconds
-    }
+function getTimeout (arc) {
+  let timeout = process.env.SANDBOX_TIMEOUT && process.env.SANDBOX_TIMEOUT * 1000
+  timeout = timeout || 5 * 1000 // 5 seconds in milliseconds
+  let ttl = arc && arc.aws.find(tuple => tuple.includes('timeout'))
+  // valid ttl is between 3 and 900 seconds (15 minutes)
+  if (ttl && ttl[1] > 3 && ttl[1] < 900) {
+    timeout = ttl[1] * 1000 // ttl seconds in milliseconds
   }
+  return timeout
+}
+
+function runInNode (options, timeout, callback) {
+  let runScript = handleNodeScript.bind(null, options, timeout, callback)
+  readFile(nodeRuntimeScriptPath, 'utf8', runScript)
+}
+
+function handleNodeScript (options, timeout, callback, err, data) {
+  if (err) {
+    // if we've gotten this far and no node script file is found go boom
+    throw err
+  } else {
+    let command = 'node'
+    let script = data.toString()
+    let args = ['-e', minify(script)]
+    spawnChild(command, args, options, timeout, callback)
+  }
+}
 
+function runInPython (options, timeout, callback) {
+  let runScript = handlePythonScript.bind(null, options, timeout, callback)
+  readFile(pythonRuntimeScriptPath, 'utf8', runScript)
+}
+
+function handlePythonScript (options, timeout, callback, err, data) {
+  if (err) {
+    // if we've gotten this far and no python script file is found go boom
+    throw err
+  }
+  let command = 'python3'
+  let script = `"${data.toString()}"`
+  let args = ['-c', script]
+  spawnChild(command, args, options, timeout, callback)
+}
+
+function spawnChild (command, args, options, timeout, callback) {
+  let cwd = options.cwd
+  let timedout = false
   // run the show
-  let child = spawn(command, options, args)
+  let child = spawn(command, args, options)
```

Comment:
These argument names being flipped was messing with my head while i was using the nodejs docs
https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options

Reply:
👀